src/Tribe/Aggregator/Cron.php

```diff
@@ -385,6 +387,81 @@ public function verify_child_record_creation() {
 		}
 	}
 
+	/**
+	 * Start the processing of the scheduled imports created with batch pushing the cron job would select and start
+	 * the beginning of the batch delivery.
+	 *
+	 * @since TBD
+	 */
+	public function start_batch_pushing_records() {
+		if ( ! tribe( 'events-aggregator.main' )->is_service_active() ) {
+			return;
+		}
+
+		$records = Tribe__Events__Aggregator__Records::instance();
+
+		$query = $records->query(
+			[
+				'post_status'    => Tribe__Events__Aggregator__Records::$status->pending,
+				'posts_per_page' => 250,
+				'order'          => 'ASC',
+				'meta_query'     => [
+					'origin-not-csv'               => [
+						'key'     => '_tribe_aggregator_origin',
+						'value'   => 'csv',
+						'compare' => '!=',
+					],
+					'batch-push-support-specified' => [
+						'key'     => '_tribe_aggregator_allow_batch_push',
+						'value'   => true,
+						'compare' => '=',
+					],
+					'batch-not-queued'             => [
+						'key'     => '_tribe_aggregator_batch_started',
+						'value'   => 'bug #23268',
+						'compare' => 'NOT EXISTS',
+					],
+				],
+				'after'          => '-4 hours',
+			]
+		);
+
+		if ( ! $query->have_posts() ) {
+			tribe( 'logger' )->log_debug( 'No Pending Batch to be started', 'EA Cron' );
+
+			return;
+		}
+
+		tribe( 'logger' )->log_debug( "Found {$query->found_posts} records", 'EA Cron' );
+
+		$cleaner = new Tribe__Events__Aggregator__Record__Queue_Cleaner();
+		foreach ( $query->posts as $post ) {
+			$record = $records->get_by_post_id( $post );
+
+			if ( null === $record || tribe_is_error( $record ) ) {
+				continue;
+			}
+
+			$cleaner->set_stall_limit( HOUR_IN_SECONDS * 22 )->set_time_to_live( HOUR_IN_SECONDS * 23 );
+
+			$cleaner->remove_duplicate_pending_records_for( $record );
+			$failed = $cleaner->maybe_fail_stalled_record( $record );
+
+			if ( $failed ) {
+				tribe( 'logger' )->log_debug( sprintf( 'Stalled record (%d) was skipped', $record->id ), 'EA Cron' );
+				continue;
+			}
+
+			// Just double Check for CSV.
+			if ( 'csv' === $record->origin ) {
```

Comment:
Should this check be done before? We'll know if a record origin is CSV right after we've checked if the record is valid, and it does not make sense to try and run the cleaning on something that should be skipped to being with.

Reply:
Looks like a great point. I'm going to do it on the other method used for the regular type of imports as well.